### ISSUE TITLE
Error obtaining codeAction's from java language server

### ISSUE THREAD

**Issue by blafond** [TypeFox/monaco-languageclient#217]
I've got a Java-based DDL/SQL language server which I can test in Eclipse IDE launcher that correctly provides quick fixes. Applying them in the text editor or via Problems view works fine.

My web-based react/patternfly app uses monaco editor and monaco-languageclient. I'm seeing in my server log that the TextDocumentService.codeActions() is correctly returning a response containing the QuickFix, however, the web console is logging an error in vscode (I think) 

With version 0.10.2 I'm getting:  **providedCodeActions.actions is undefined**
With version 0.12.0 I get:  **r.actions is undefined**

appears to be the same issue, just more obfuscated text.  I've debugged down into monaco-editor's codeAction.js/getCodeActions() method but I can't track the request/response.

Is there any set-up/properties I need to add/set to enable the client to process the codeActions or anything else I could check?

**Comment by apupier** [comment]
this log gives an idea of the message sent by the server to the Monaco Client:

```
14:19:24.771 [Thread-186] INFO  i.s.d.l.c.InvalidDatatypeQuickFix -     createCodeAction() action text = Did you mean string datatype?
14:19:24.771 [Thread-186] INFO  i.s.d.l.c.InvalidDatatypeQuickFix -     returning  Action = CodeAction [
  title = "Did you mean string datatype?"
  kind = "quickfix"
  diagnostics = SingletonList (
    Diagnostic [
      range = Range [
        start = Position [
          line = 1
          character = 13
        ]
        end = Position [
          line = 1
          character = 16
        ]
      ]
      severity = Error
      code = null
      source = null
      message = "Invalid datatype: 'foo' at: Line 2 Column 14"
      relatedInformation = null
    ]
  )
  edit = WorkspaceEdit [
    changes = {inmemory://model/1=[TextEdit [
      range = Range [
        start = Position [
          line = 1
          character = 13
        ]
        end = Position [
          line = 1
          character = 16
        ]
      ]
      newText = "string"
    ]]}
    documentChanges = null
    resourceChanges = null
  ]
  command = null
]
14:19:24.771 [Thread-186] INFO  i.s.d.l.TeiidDdlTextDocumentService -     RETURNING InvalidDatatypeQuickFix
```

**Comment by apupier** [comment]
thinking loud:

- Can it be that Monaco Language Client is expecting an Array, although a single element is provided? https://github.com/TypeFox/monaco-languageclient/blob/fb587b3ff258d48194c8667a6ade2d0fd87d06bd/client/src/monaco-converter.ts#L568
- Can it be related to this code with a fixme https://github.com/TypeFox/monaco-languageclient/blob/fb587b3ff258d48194c8667a6ade2d0fd87d06bd/client/src/monaco-languages.ts#L251-L252 ? the pointed issue int eh fixme has been fixed.
- @blafond Can you try to return 2 CodeActions? it is to check that it is not a single-element list that is nto converted to a single element directly and is not handled everywhere.

**Comment by apupier** [comment]
this issue is mentioning "parameter actions undefined value", so looks very similar. it is already fixed https://github.com/TypeFox/monaco-languageclient/issues/154 so surely another specific case?

**Comment by apupier** [comment]
most (all?) of the other converters for lists have a null check:
https://github.com/TypeFox/monaco-languageclient/blob/fb587b3ff258d48194c8667a6ade2d0fd87d06bd/client/src/monaco-converter.ts#L556-L559
https://github.com/TypeFox/monaco-languageclient/blob/fb587b3ff258d48194c8667a6ade2d0fd87d06bd/client/src/monaco-converter.ts#L542-L545
https://github.com/TypeFox/monaco-languageclient/blob/fb587b3ff258d48194c8667a6ade2d0fd87d06bd/client/src/monaco-converter.ts#L532-L535
https://github.com/TypeFox/monaco-languageclient/blob/fb587b3ff258d48194c8667a6ade2d0fd87d06bd/client/src/monaco-converter.ts#L661-L664

There was a PR https://github.com/TypeFox/monaco-languageclient/pull/159/files to do the same for codeActions but it has never been merged.

Could it be the root issue?

**Comment by apupier** [comment]
I created a [PR](https://github.com/TypeFox/monaco-languageclient/pull/221) to check for nulliness. @blafond can you try with a build from the version of this branch please?

**Comment by blafond** [comment]
I looked at your PR on monaco-converter.ts and compared it against the class in my workspace.  First, my local library had a monaco-converter.js and wasn't anywhere close to the same code as latest release.  So I updated my version of the monaco-languageclient to latest:  0.13.0 and monaco-editor-core: 0.20.0 and now codeActions are being generated and accepted by the UI.

I didn't run against your branch version or PR though.
